### PR TITLE
fixes null reference created upon player death

### PR DESCRIPTION
**Pull request by gdiago**
Fixes null reference exception caused when a player is killed.

### CODE REVIEW

**Review by wynn-rj — APPROVED** [review]

File: Assets/Scripts/Battlefield/Camera/CameraTarget.cs [hunk 1]
```diff
@@ -19,7 +19,11 @@ void Start()
     
         void Update()
         {
-            transform.position = player.position;
+            if(player != null)
```
Comment (subashch6) [inline Assets/Scripts/Battlefield/Camera/CameraTarget.cs:22]: I think this camera should be destroyed as well when the player dies so there arent extra game object. Could you implement a function or apply an abstract class that units or creatures derive from that have an OnDeath() or something to kill destroy this camera along with itself.

**Review by subashch6 — CHANGES_REQUESTED** [review]
Destroy camera in an OnDeath() function. Better if done from an Abstract class that unit or creature is derived from.

File: Assets/Scripts/Battlefield/Camera/CameraTarget.cs (hunk 1 above)
Comment (gdiago) [reply Assets/Scripts/Battlefield/Camera/CameraTarget.cs:22]: I didn't write these classes so I'm not entirely sure how Jack has them set up. I just spotted the bug and made a quick fix, so Jack will probably have to do that

Comment (gdiago) [reply Assets/Scripts/Battlefield/Camera/CameraTarget.cs:22]: It's only 3 lines of code so we should just merge it in so we aren't getting null reference exceptions for timebox 2 then worry about destroying the cameras for timebox 3

Comment (gdiago) [reply Assets/Scripts/Battlefield/Camera/CameraTarget.cs:22]: It's also still best practice IMO to check if an object exists before trying to reference it anyway, so I think we should leave this check and then add the code for destroying cameras later on

Comment (subashch6) [reply Assets/Scripts/Battlefield/Camera/CameraTarget.cs:22]: Sorry, i didnt see this earlier. Yes, I think it should be ok for now. I will accept.

**Review by subashch6 — APPROVED** [review]

**Pull request merged**